src/workerd/api/pyodide/pyodide.h

```diff
@@ -500,15 +501,28 @@ kj::Promise<void> loadPyodidePackage(const PythonConfig& pyConfig,
     const PyodidePackageManager& pyodidePackageManager,
     kj::StringPtr packagesVersion,
     kj::StringPtr filename,
-    kj::HttpClient& client,
-    kj::Timer& timer,
-    kj::HttpHeaderTable& table);
+    kj::Network& network,
+    kj::Timer& timer);
 
 // Preloads all required Python packages for a worker
-void fetchPyodidePackages(const PythonConfig& pyConfig,
+kj::Promise<void> fetchPyodidePackages(const PythonConfig& pyConfig,
     const PyodidePackageManager& pyodidePackageManager,
     kj::ArrayPtr<kj::String> pythonRequirements,
-    workerd::PythonSnapshotRelease::Reader pythonSnapshotRelease);
+    workerd::PythonSnapshotRelease::Reader pythonSnapshotRelease,
+    kj::Network& network,
+    kj::Timer& timer);
+
+// Preload Pyodide bundle during workerd startup
+kj::Promise<kj::Maybe<jsg::Bundle::Reader>> fetchPyodideBundle(
+    const PythonConfig& pyConfig, kj::String version, kj::Network& network, kj::Timer& timer);
+
+// Helper functions for bundle file operations
+kj::Path getPyodideBundleFileName(kj::StringPtr version);
```

Comment:
nit: any function that isn't really used by including files doesn't need to be in the header.